src/XrdHttp/XrdHttpReq.cc

```diff
@@ -749,6 +749,15 @@ void XrdHttpReq::addCgi(const std::string &key, const std::string &value) {
   hdr2cgistr.append(value);
 }
 
+std::unique_ptr<XrdOucEnv> XrdHttpReq::parseDestination() {
```

Comment:
@bbockelm Any suggestions for this parse function? 
 - Should there be a member `destination_opaque` in XrdHttpReq as opposed to returning something - my idea is to not bloat the object unless we need the parsed data again. 
 - I opted to keep it as a member of XrdHttpReq as opposed to something within XrdHttpUtils

Reply:
This is too heavy a hammer for my taste.

Fundamentally, you really just need to know if `authz` is present in the URL.  Parsing this to a hash map of strings-to-strings is an expensive operation with multiple allocations and copies.

[I'd prefer an approach like this one](https://github.com/PelicanPlatform/xrdcl-pelican/blob/main/src/XrdClS3/XrdClS3Factory.cc#L291-L318) where you just step through the string piece by piece and return true/false depending on the presence of `authz`.